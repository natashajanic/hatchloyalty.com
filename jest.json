{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
