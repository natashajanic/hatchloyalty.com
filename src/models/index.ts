import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
