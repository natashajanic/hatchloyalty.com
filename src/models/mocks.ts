import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
