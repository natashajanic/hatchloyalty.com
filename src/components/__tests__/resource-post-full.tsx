import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import { mockResourcePost } from 'src/models/mocks'
import ResourcePostFull, { IResourcePostFullProps } from '../resource-post-full'

describe('ResourcePostFull', () => {
  let props: IResourcePostFullProps
  let mountedComponent: ShallowWrapper<any, any, ResourcePostFull>

  beforeEach(() => {
    props = {
      post: mockResourcePost,
    }
    mountedComponent = shallow(<ResourcePostFull {...props} />)
  })

  describe('render', () => {
    it('renders the resource title', async () => {
      expect(mountedComponent.text()).toContain(
        mockResourcePost.frontmatter.title
      )
    })

    it("renders the raw post's body as HTML", async () => {
      expect(mountedComponent.html()).toContain(mockResourcePost.html)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
