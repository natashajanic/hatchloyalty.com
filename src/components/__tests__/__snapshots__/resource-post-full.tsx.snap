// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcePostFull render matches the expected snapshot 1`] = `
<div>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    py={4}
  >
    <Styled(Component)
      blacklist={
        Array [
          "css",
          "border",
          "borderTop",
          "borderRight",
          "borderBottom",
          "borderLeft",
          "borderColor",
          "fontSize",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
        ]
      }
      borderBottom="1px solid"
      borderColor="grayLight"
      fontSize={5}
      is="h1"
      m={0}
      pb={4}
    >
      Test Resource Post
    </Styled(Component)>
  </Styled(Component)>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>Test resource post body content.</div>",
      }
    }
  />
</div>
`;
